app/src/main/java/woowacourse/movie/feature/movieSelection/adapter/MovieAdapter.kt

```
@@ -0,0 +1,66 @@
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int,
+    ): RecyclerView.ViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
+
+        return when (viewType) {
+            VIEW_TYPE_MOVIE -> MovieViewHolder(view, onReservationClick)
+            VIEW_TYPE_AD -> AdViewHolder(view)
+            else -> throw IllegalStateException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)